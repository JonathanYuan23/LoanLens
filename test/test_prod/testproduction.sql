-- userAllLoan.sql
SELECT 
    reason,
    loan_amount,
    balance_paid,
    date_created
FROM 
    Loans
WHERE 
    user_id = 22;

-- delimiter
--------------------------------------------------------------

-- userAsset.sql
SELECT 
    Assets.asset_id,
    Assets.asset_type,
    Assets.worth
FROM 
    Assets
JOIN 
    AssetToOwner ON Assets.asset_id = AssetToOwner.asset_id
WHERE 
    AssetToOwner.user_id = 21;

-- delimiter
--------------------------------------------------------------

-- userAssetValue.sql
SELECT u.name, SUM(a.worth) AS total_assets
FROM Users u
LEFT JOIN AssetToOwner ao ON u.user_id = ao.user_id
LEFT JOIN Assets a ON ao.asset_id = a.asset_id
WHERE u.user_id = 21
GROUP BY u.user_id;

-- delimiter
--------------------------------------------------------------

-- userHousehold.sql
WITH RECURSIVE HouseholdMembers AS (
    -- Anchor member: the given user
    SELECT 
        user_id 
    FROM 
        Users 
    WHERE 
        user_id = 4
    UNION
    -- Get dependents and providers
    SELECT 
        d.dependant_id AS user_id 
    FROM 
        Dependant d
    JOIN 
        HouseholdMembers hm ON d.provider_id = hm.user_id
    UNION
    SELECT 
        d.provider_id AS user_id 
    FROM 
        Dependant d
    JOIN 
        HouseholdMembers hm ON d.dependant_id = hm.user_id
    UNION
    -- Get spouses
    SELECT 
        m.spouse_id_2 AS user_id 
    FROM 
        Married m 
    JOIN 
        HouseholdMembers hm ON m.spouse_id_1 = hm.user_id
    UNION
    SELECT 
        m.spouse_id_1 AS user_id 
    FROM 
        Married m 
    JOIN 
        HouseholdMembers hm ON m.spouse_id_2 = hm.user_id
)

SELECT DISTINCT hm.user_id, u.name
FROM HouseholdMembers hm
JOIN Users u ON hm.user_id = u.user_id;


-- delimiter
--------------------------------------------------------------

-- userHouseholdIncome.sql
WITH RECURSIVE HouseholdMembers AS (
    -- Anchor member: the given user
    SELECT 
        user_id 
    FROM 
        Users 
    WHERE 
        user_id = 6

    UNION

    -- Get dependents and providers
    SELECT 
        d.dependant_id AS user_id 
    FROM 
        Dependant d
    JOIN 
        HouseholdMembers hm ON d.provider_id = hm.user_id

    UNION

    SELECT 
        d.provider_id AS user_id 
    FROM 
        Dependant d
    JOIN 
        HouseholdMembers hm ON d.dependant_id = hm.user_id

    UNION

    -- Get spouses
    SELECT 
        m.spouse_id_2 AS user_id 
    FROM 
        Married m 
    JOIN 
        HouseholdMembers hm ON m.spouse_id_1 = hm.user_id

    UNION

    SELECT 
        m.spouse_id_1 AS user_id 
    FROM 
        Married m 
    JOIN 
        HouseholdMembers hm ON m.spouse_id_2 = hm.user_id
)

SELECT 
    SUM(u.income) AS total_household_income
FROM 
    HouseholdMembers hm
JOIN 
    Users u ON hm.user_id = u.user_id;

-- delimiter
--------------------------------------------------------------

-- userLoanInfo.sql
SELECT user_id, SUM(loan_amount) AS total_loan_amount, SUM(balance_paid) AS total_balance_paid
FROM Loans
WHERE user_id = 14
GROUP BY user_id;

