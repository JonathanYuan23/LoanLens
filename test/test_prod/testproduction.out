reason, loan_amount, balance_paid, date_created
Investment Opportunities, 180329.00, 126560.00, 2013-06-09 00:00:00

asset_id, asset_type, worth
9, house, 1142018.00
10, car, 80270.00

name, total_assets
Michael Patel, 1222288.00

user_id
6
1
10
2
3
4
5
7
9

total_household_income
324292.00

user_id, total_loan_amount, total_balance_paid
14, 152405.00, 50211.00
